src/lib/checkWikiValidity.ts

```diff
@@ -1,261 +0,0 @@
-			media.source === MediaSource.IPFS_VID
-		) {
-			return media.id.length === IPFS_HASH_LENGTH;
-		}
-
-		if (media.source === MediaSource.YOUTUBE) {
-			const youtubePattern =
-				/^.*(?:youtu\.be\/|v\/|vi\/|u\/\w\/|embed\/|shorts\/|watch\?v=)([^#&?]*).*/;
-			return (
-				media.id === `https://www.youtube.com/watch?v=${media.name}` &&
-				youtubePattern.test(media.id)
-			);
-		}
-
-		if (media.source === MediaSource.VIMEO) {
-			return media.id === `https://vimeo.com/${media.name}`;
-		}
-
-		return media.type ? Object.values(MediaType).includes(media.type) : true;
-	});
-
-	const iconMediaCount = wiki.media.filter(
-		(media) => media.type === MediaType.ICON,
-	).length;
-
-	return (
-		wiki.media.length <= MAX_MEDIA_COUNT &&
-		isMediaContentValid &&
-		iconMediaCount <= 1
-	);
-};
-
-/**
- * Checks if any media in the wiki is still uploading.
- * @param wiki - The wiki object to check.
- * @returns True if any media is still uploading, false otherwise.
- */
-export const isAnyMediaUploading = (wiki: Wiki) =>
-	wiki.media?.some((media) => media.id.endsWith(MEDIA_UPLOAD_PENDING_SUFFIX)) ??
-	false;
-
-/**
- * Checks if the event URL is missing for an event wiki.
- * @param wiki - The wiki object to check.
- * @returns True if the event URL is missing, false otherwise.
- */
-export const isEventUrlMissing = (wiki: Wiki) => {
-	if (wiki.tags.some((tag) => tag.id === "Events")) {
-		const referencesData =
-			wiki.metadata.find((meta) => meta.id === CommonMetaIds.REFERENCES)
-				?.value || "[]";
-		const references = JSON.parse(referencesData);
-		return !references.some(
-			(item: { description: string }) =>
-				item.description.toLowerCase() === "event link",
-		);
-	}
-	return false;
-};
-
-/**
- * Checks if the event date is missing for an event wiki.
- * @param wiki - The wiki object to check.
- * @returns True if the event date is missing, false otherwise.
- */
-export const isEventDateMissing = (wiki: Wiki) =>
-	wiki.tags.some((tag) => tag.id === "Events") && wiki.events?.length === 0;
-
-/**
- * Checks if the wiki summary exceeds the maximum length.
- * @param wiki - The wiki object to check.
- * @returns True if the summary exceeds the limit, false otherwise.
- */
-export const isSummaryTooLong = (wiki: Wiki) =>
-	!!(wiki.summary && wiki.summary.length > WIKI_SUMMARY_MAX_LENGTH);
-
-/**
- * Checks if the wiki has no citations.
- * @param wiki - The wiki object to check.
- * @returns True if there are no citations, false otherwise.
- */
-export const hasNoCitations = (wiki: Wiki) => {
-	const references = wiki.metadata.find(
-		(meta) => meta.id === CommonMetaIds.REFERENCES,
-	);
-	return !references?.value || references.value.length === 0;
-};
-
-/**
- * Validates a wiki object and returns the validation result.
- * @param wiki - The wiki object to validate.
- * @returns An object containing the validation result and an error message if applicable.
- */
-export const validateWiki = (wiki: Wiki) => {
```

Comment:
@Adebesin-Cell Lets keep this until we fully migrate to zod schema, at start we can just integrate this in our ai editor

Reply:
Oh alright! I'll revert it, I believe we can open sub issues to use version 2 of iq-utils in new project, instead of the old one, what do you think?